src/Conda.jl

```diff
@@ -137,34 +187,40 @@ function _install_conda(force=false)
             end
         end
         Conda.add_channel("defaults")
+        # Update conda because conda 4.0 is needed and miniconda download installs only 3.9
+        run(_set_conda_env(`$(Conda.conda) update -y conda`))
+    end
+    if !isdir(prefix(env))
+        # conda doesn't allow totally empty environments. using zlib as the default package
+        run(_set_conda_env(`$(conda_bin(RootEnv)) create -y -p $(prefix(env)) zlib`, RootEnv))
     end
 end
 
 "Install a new package."
-function add(pkg::AbstractString)
-    _install_conda()
-    run(_set_conda_env(`$conda install -y $pkg`))
+function add(pkg::AbstractString, env::Environment=RootEnv)
+    _install_conda(false, env)
```

Comment:
The `false` parameter should not be needed with the default parameters. You can inverse the parameter order in `_install_conda` to be `_install_conda(env::Environment, force=false)`. This way you do not even need the default value for `env`, which will be provided by all the callers.